### ISSUE TITLE
DNS are not being queried by Adguard Home

### ISSUE THREAD

**Issue by SeriousHoax** [AdguardTeam/AdGuardHome#4757]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

* **Version of AdGuard Home server:**
  * v0.107.8
* **How did you install AdGuard Home:**
  * GitHub releases on Windows 11 Beta
* **How did you setup DNS configuration:**
  * System
* **If it's a router or IoT, please write device model:**
  * 
* **CPU architecture:**
  * AMD64
* **Operating system and version:**
  * Windows 11 Beta 22622.290

### Expected Behavior

DNS queries should be done by Adguard Home
### Actual Behavior

DNS queries are being performed by my router's DNS instead. I've used Adguard Home in the past so I know how to set it up. So pretty sure everything is fine in this regard. Twice it worked for a few queries, but then it stopped working even though Adguard service was running just fine.
### Screenshots

<details><summary>Screenshot:</summary>

</details>

**Comment by cyb3rz3us** [comment]
1. Have you confirmed your Windows PC has your AGH server IP cfg'd as the DNS server?
2. How do you know the router is doing the lookup and AGH isn't?
3. If the PC is sending DNS queries to the AGH server (presuming Q1 = YES), why does the router even begin a DNS query?
4. Is it possible some other device is sending DNS queries to the router?
5. Is AGH cfg'd to use the router as the upstream DNS?

**Comment by SeriousHoax** [comment]
.

> 1. Have you confirmed your Windows PC has your AGH server IP cfg'd as the DNS server?
> 2. How do you know the router is doing the lookup and AGH isn't?
> 3. If the PC is sending DNS queries to the AGH server (presuming Q1 = YES), why does the router even begin a DNS query?
> 4. Is it possible some other device is sending DNS queries to the router?
> 5. Is AGH cfg'd to use the router as the upstream DNS?

Hello,
1. Yes, I have. Also tried the other IP4 IP, but nothing changed.
<details><summary>Screenshot:</summary>

![2](https://user-images.githubusercontent.com/29574622/179417105-2755b231-1c01-4bbe-a5d5-91afb5592162.png)
![3](https://user-images.githubusercontent.com/29574622/179417109-e1fd1802-e540-41e1-a472-ce1e7317d5d6.png)
</details><br/>

2. I see it on any dns test website. My router has NextDNS as the DNS and that's what I see in tests even after enabling Adguard Home. Also I can see NextDNS working by logging into my NextDNS account and of course in Adguard Home query logs I can see that query logs are not generating logs.

<details><summary>Screenshot:</summary>

![1](https://user-images.githubusercontent.com/29574622/179417150-0c3fb52d-201d-439c-adab-42980c2186f1.png)
</details><br/>

3. I don't know about that. I also tried YogaDNS app which worked perfectly without any issue.
4. No, because Adguard Home is running on my Windows 11 only.

5. No. I tried different DNS providers, but the result is the same. I also allowed inbound connection for "AdGuardHome.exe". Don't know if that's required. But just in case. I'm using Windows Firewall in default mode. 
<details><summary>Screenshot:</summary>

![5](https://user-images.githubusercontent.com/29574622/179417245-d35cc9ca-445c-457d-a054-3a5e786a1451.png)
![6](https://user-images.githubusercontent.com/29574622/179417251-41cd4a87-155e-4c22-a436-73a4f7931861.png)
</details><br/>

Sometimes but rarely both ADH and router dns works and this is what dnsleak shows in that case.

<details><summary>Screenshot:</summary>

![4](https://user-images.githubusercontent.com/29574622/179417332-e4ce3025-161e-4462-b87e-13f27a721c02.png)

</details><br/>

**Comment by cyb3rz3us** [comment]
It looks like you are using the AG client on a Windows machine, correct?

If so, then I think you are on the wrong repo - pretty sure you want: https://github.com/AdguardTeam/AdguardForWindows

AFAIK, AGH is only available on Linux & Mac and is basically a DNS proxy on steroids.

**Comment by SeriousHoax** [comment]
> It looks like you are using the AG app on a Windows client machine, correct?
> 
> If so, then I think you are on the wrong repo - pretty sure you want: https://github.com/AdguardTeam/AdguardForWindows
> 
> AFAIK, AGH is only available on Linux & Mac and is basically a DNS proxy on steroids.

No, I'm not using Adgurad Desktop. I'm using Adguard Home and it's available for Windows as well. As I said in my issue, I have actually used Adguard Home on my PC in the past without any issue like the one I described here.
https://github.com/AdguardTeam/AdGuardHome/releases/download/v0.107.8/AdGuardHome_windows_amd64.zip

**Comment by cyb3rz3us** [comment]
> actually used Adguard Home on my PC in the past

Fair enough - my mistake.  I didn't see it listed anywhere on the main page.  

Regardless, the screenshot you added does not show any Windows IP client config.  It looks like you pasted something  from AG...

**Comment by SeriousHoax** [comment]
> Regardless, the screenshot you added isn't familiar to me - is that from Win 11? It looks like you pasted something from AG...
It's ok. No problem. Yeah, it's from Windows 11. The UI is a bit different from Windows 10, but functionality is the same. 
Disabling Firewall didn't have any effect.
My router is a fairly basic one. Only DNS is changed and UPnP is disabled. My router hasn't changed since the last time I used Adguard Home.

**Comment by cyb3rz3us** [comment]
I don't have Win11 or else I would test here but I do have Win10.  Before diving in there, let me ask, does this also fail on your end with Win10?

**Comment by SeriousHoax** [comment]
> I don't have Win11 or else I would test here but I do have Win10. Before diving in there, let me ask, does this also fail on your end with Win10?

I'm on the opposite spectrum. I can't test it on Windows 10 because I have Windows 11 only.

**Comment by cyb3rz3us** [comment]
OK - well I will still try Win10 because now I'm just curious...may not tell us much but worth a shot...

**Comment by SeriousHoax** [comment]
> OK - well I will still try Win10 because now I'm just curious...may not tell us much but worth a shot...

Thanks. Let me know what happens.

**Comment by cyb3rz3us** [comment]
FYI - plain DNS works fine on my Win10 machine.  Setting up DoH now...will update when done...

**Comment by SeriousHoax** [comment]
> #### Update
> DoH works fine also.
> 
> What do you see in your AGH query log? Anything?
> 
> #### Previous post
> FYI - plain DNS works fine on my Win10 machine. Setting up DoH now...will update when done...

Sometimes both Adguard Home's DNS and router's DNS works together. But almost all are being handled by routter. Router DNS shouldn't work at all.

**Comment by cyb3rz3us** [comment]
But I was asking if there is anything in the AGH query log?

**Comment by SeriousHoax** [comment]
> But I was asking if there is anything in the AGH query log?

As I said sometimes Adguard Home's DNS works and when it does, logs are created. But most of the time like 9/10 times logs are not created because those DNA queries are being solved by my router's DNS. So this is an unexpected behavior.

**Comment by ainar-g** [comment]
@SeriousHoax, hello.  Just to be clear, have you disabled the automatic DNS server assignment in the DHCP settings?  If not, your router's DNS could overwrite the other DNS settings.

**Comment by SeriousHoax** [comment]
> @SeriousHoax, hello.  Just to be clear, have you disabled the automatic DNS server assignment in the DHCP settings?  If not, your router's DNS could overwrite the other DNS settings.

Hi! Yes, it's set to manual. For example, if I set Cloudflare DNS on Windows then that's what it uses ignoring my router DNS. 
But when I set 127.0.0.1 as my DNS after enabling Adguard Home, it's not working as expected. Most DNS queries are being solved by my router DNS and some occasionally are being solved by the one set in Adguard Home.
BTW, when I created the issue I was using Windows 11 Beta but now I'm on the stable build and the problem remains the same. So don't know why is this happening.

**Comment by SeriousHoax** [comment]
> @SeriousHoax, can you please provide a screenshot of these settings?  Also, did you put `127.0.0.1` once or twice?  Because most OSes use a primary and a secondary DNS resolvers, and they can switch between the two at random.

I put it only once. 
Here are the screenshots:

![0.png](https://user-images.githubusercontent.com/29574622/181519537-723d6355-1167-4d80-8be7-536a107ea2a4.png)

![1.png](https://user-images.githubusercontent.com/29574622/181519565-590e5412-9602-419c-91cd-b65ecf6e05cb.png)

**Comment by ainar-g** [comment]
Please try putting the address into the *Alternate DNS* field as well.  Perhaps also restart the machine, to make sure that the network settings are applied.

**Comment by SeriousHoax** [comment]
> Please try putting the address into the *Alternate DNS* field as well.  Perhaps also restart the machine, to make sure that the network settings are applied.

Just tried both of your suggestions but still no luck. 
0 DNS queries solved by Adguard Home.

**Comment by ainar-g** [comment]
@SeriousHoax, what does `nslookup.exe example.com` show in the `Server` and `Address` fields?  Also, what does `Get-DnsClientServerAddress` show for your interface?

**Comment by SeriousHoax** [comment]
> Get-DnsClientServerAddress

This is what it shows. Is there anything wrong here?
![1](https://user-images.githubusercontent.com/29574622/181921183-6ffb663d-a0c7-466e-b819-a65760563533.png)
![2](https://user-images.githubusercontent.com/29574622/181921189-9e00f736-79f0-48df-abb8-7d9d349be965.png)

**Comment by ainar-g** [comment]
@SeriousHoax, apologies for the long lack of response.  I see that you have a ProtonVPN interface.  VPNs commonly use their own DNS servers, so it is possible that the VPN is what is blocking you from using AGH.  You can either disable the VPN to see if the queries go to AGH, or look through the VPN settings to see if you can set AGH as the DNS server.

**Comment by SeriousHoax** [comment]
> @SeriousHoax, apologies for the long lack of response. I see that you have a ProtonVPN interface. VPNs commonly use their own DNS servers, so it is possible that the VPN is what is blocking you from using AGH. You can either disable the VPN to see if the queries go to AGH, or look through the VPN settings to see if you can set AGH as the DNS server.

Hi! The ProtonVPN interface is present there even when the VPN is not running. But anyway, since my last comment, I reinstalled Windows 11 22H2 and didn't install ProtonVPN on my system. But still, Adguard Home doesn't work for me. It's the same as before.

**Comment by ainar-g** [comment]
I see, thanks for the update.  Does the output of `Get-DnsClientServerAddress` show only AdGuard Home IPs for your internet connection now?

**Comment by SeriousHoax** [comment]
> I see, thanks for the update. Does the output of `Get-DnsClientServerAddress` show only AdGuard Home IPs for your internet connection now?

Thanks. It's working now. I even had to enable IP6 on the network setting and put `::1` as DNS. I never had to do this before in Windows 10 so it's a bit weird. For normal usage if I'm not running Adguard Home then it's not need to set IP6 DNS as my internet provider don't support IP6.
